nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/NuxeoRequestControllerFilter.java

```diff
@@ -263,8 +263,12 @@ public static boolean simpleReleaseSyncOnSession(HttpServletRequest request) {
     protected void addHeaders(HttpServletRequest request, HttpServletResponse response,
             RequestFilterConfig config) {
         addConfiguredHeaders(response);
-        if (request.getMethod().equals("GET") && config.isCached()) {
-            addCacheHeaders(response, config.isPrivate(), config.getCacheTime());
+        if (request.getMethod().equals("GET")) {
+            if (config.isCached()) {
+                addCacheHeaders(response, config.isPrivate(), config.getCacheTime());
+            } else if (config.isPrivate()) {
+                response.setHeader("Cache-Control", "private, no-cache, no-store");
```

Comment:
Could you move all that logic into `addCacheHeaders` please?

Reply:
would need to pass config.isCached() too so might as well call `addCacheHeaders` for both cached and uncached reponses

Reply:
I'm ok with passing the whole `config` to `addCacheHeaders` yes and doing all this header setting there.

Reply:
👍